File: app/src/main/java/com/rachelleignacio/listbucket/domain/interactors/GetAllListItemsInteractor.java
package com.rachelleignacio.listbucket.domain.interactors;

import com.rachelleignacio.listbucket.domain.interactors.base.Interactor;
import com.rachelleignacio.listbucket.domain.models.ListItem;

import java.util.List;

/**
 * Created by rachelleignacio on 3/9/17.
 */

public interface GetAllListItemsInteractor extends Interactor {
    interface Callback {
        void onListItemsRetrieved(List<ListItem> items);
    }
}

File: app/src/main/kotlin/com/rachelleignacio/listbucket/domain/interactors/GetAllListItemsInteractor.kt
package com.rachelleignacio.listbucket.domain.interactors

import com.rachelleignacio.listbucket.domain.interactors.base.Interactor
import com.rachelleignacio.listbucket.domain.models.ListItem

/**
 * Created by rachelleignacio on 8/9/17.
 */
interface GetAllListItemsInteractor : Interactor {
    interface Callback {
        fun onListItemsRetrieved(items: MutableList<ListItem>)
    }
}